### PR TITLE
Fixes several pointer-generator lstm bugs

### PR DESCRIPTION
**Pull request by Adamits**
I was not able to run the pointer-generator LSTM. It looks like the breaks have been here since #72.

- We now actually pass the encoded features through to the decode_step
- We correctly concatenate the feature attention if provided (previously we erroneously concatenated the source attention * 2)
- No longer adds the extra breaking dimension to the generation probability.

### CODE REVIEW

File: yoyodyne/models/pointer_generator.py [hunk 1]
```diff
@@ -259,7 +259,7 @@ def decode_step(
                 last_h0.transpose(0, 1), features_enc, features_mask
             )
             # -> B x 1 x 4*hidden_size.
-            context = torch.cat([context, context], dim=2)
+            context = torch.cat([context, features_context], dim=2)
```
Comment (kylebgorman) [inline yoyodyne/models/pointer_generator.py:262]: Driveby, but would it make sense to call them `source_context` and `features_context`?

**Review by kylebgorman — APPROVED** [review]
LGTM.

File: yoyodyne/models/pointer_generator.py (hunk 1 above)
Comment (Adamits) [reply yoyodyne/models/pointer_generator.py:262]: True. Can do

Comment (kylebgorman) [reply yoyodyne/models/pointer_generator.py:262]: Okay, let me know when that's done and I'll merge.

**Comment by kylebgorman** [comment]
I just tested some Maori and Polish (the latter has features) and it seems to be working also. Merging now.
